plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java

```diff
@@ -107,7 +109,7 @@ protected Control createDialogArea(final Composite parent) {
           @Override
           public int getMessageType(ValidationStatusProvider statusProvider) {
             int type = super.getMessageType(statusProvider);
-            setValid(type != IMessageProvider.ERROR);
+            setValid(type != IMessageProvider.ERROR && content.hasSelection());
```

Comment:
@briandealwis Isn't this by-passing or abusing databinding? I feel if the project selector does not have any selection, it should give the ERROR state with the message "select a project", which I believe was the behavior prior to the new project selector.